crates/ruff/src/directives.rs

```diff
@@ -429,6 +454,50 @@ ghi
             NoqaMapping::from_iter([TextRange::new(TextSize::from(6), TextSize::from(28))])
         );
 
+        let contents = "x = f'abc {
+a
+    *
+        b
+}'
+y = 2
+";
+        assert_eq!(
+            noqa_mappings(contents),
+            NoqaMapping::from_iter([TextRange::new(TextSize::from(0), TextSize::from(32))])
+        );
+
+        let contents = "x = f'''abc
+def
+ghi
+'''
+y = 2
+z = x + 1";
+        assert_eq!(
+            noqa_mappings(contents),
+            NoqaMapping::from_iter([TextRange::new(TextSize::from(0), TextSize::from(23))])
+        );
+
+        let contents = "x = 1
+y = f'''abc
+def
+ghi
+'''
+z = 2";
+        assert_eq!(
+            noqa_mappings(contents),
+            NoqaMapping::from_iter([TextRange::new(TextSize::from(6), TextSize::from(29))])
+        );
+
+        let contents = "x = 1
+y = f'''abc
```

Comment:
Can you add an example for nested f-strings

Reply:
Added a test case.